x/sync/syncable_db_client.go

```diff
@@ -0,0 +1,141 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package sync
+
+import (
+	"context"
+	"errors"
+
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/x/merkledb"
+
+	syncpb "github.com/ava-labs/avalanchego/proto/pb/sync"
+)
+
+var _ SyncableDB = (*SyncableDBClient)(nil)
+
+func NewSyncableDBClient(client syncpb.SyncableDBClient) *SyncableDBClient {
+	return &SyncableDBClient{client: client}
+}
+
+type SyncableDBClient struct {
+	client syncpb.SyncableDBClient
+}
+
+func (c *SyncableDBClient) GetMerkleRoot(ctx context.Context) (ids.ID, error) {
+	resp, err := c.client.GetMerkleRoot(ctx, &emptypb.Empty{})
+	if err != nil {
+		return ids.ID{}, err
+	}
+	return ids.ToID(resp.RootHash)
+}
+
+func (c *SyncableDBClient) GetChangeProof(
+	ctx context.Context,
+	startRootID ids.ID,
+	endRootID ids.ID,
+	startKey []byte,
+	endKey []byte,
+	keyLimit int,
+) (*merkledb.ChangeProof, error) {
+	resp, err := c.client.GetChangeProof(ctx, &syncpb.GetChangeProofRequest{
+		StartRootHash: startRootID[:],
+		EndRootHash:   endRootID[:],
+		StartKey:      startKey,
+		EndKey:        endKey,
+		KeyLimit:      uint32(keyLimit),
+	})
+	if err != nil {
+		return nil, err
+	}
+	var proof merkledb.ChangeProof
+	if err := proof.UnmarshalProto(resp); err != nil {
+		return nil, err
+	}
+	return &proof, nil
+}
+
+func (c *SyncableDBClient) VerifyChangeProof(
+	ctx context.Context,
+	proof *merkledb.ChangeProof,
+	startKey []byte,
+	endKey []byte,
+	expectedRootID ids.ID,
+) error {
+	resp, err := c.client.VerifyChangeProof(ctx, &syncpb.VerifyChangeProofRequest{
+		Proof:            proof.ToProto(),
+		StartKey:         startKey,
+		EndKey:           endKey,
+		ExpectedRootHash: expectedRootID[:],
+	})
+	if err != nil {
+		return err
+	}
+
+	// TODO there's probably a better way to do this.
```

Comment:
I think if we moved errors into grpc status the below is handled by the nil check on error above. See syncable server for more details